utils/kernel.c

```diff
@@ -465,6 +465,46 @@ static int __setup_kernel_tracing(struct uftrace_kernel_writer *kernel)
 	return -EINVAL;
 }
 
+static void skip_kernel_functions(struct uftrace_kernel_writer *kernel)
+{
+	unsigned int i;
+	struct kfilter *kfilter;
+	const char *skip_funcs[] = {
+		"syscall_trace_enter_phase1",
+		"smp_irq_work_interrupt",
+		"syscall_slow_exit_work",
+	};
+
+	for (i = 0; i < ARRAY_SIZE(skip_funcs); i++) {
+		bool found = false;
+		const char *name = skip_funcs[i];
+		struct kfilter *pos, *tmp;
+
+		/* Don't skip it if user particularly want to see them*/
+		list_for_each_entry_safe(pos, tmp, &kernel->filters, list) {
+			if (strcmp(pos->name, name))
+				goto found;
+		}
+
+		list_for_each_entry_safe(pos, tmp, &kernel->patches, list) {
```

Comment:
You don't need to use the "_safe" version unless you want to delete an entry.  Same goes to the loop above.